File: practica1/Ejercicio1/src/ejercicio1/FactoriaMontana.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ejercicio1;

/**
 *
 * @author carmenxufdz
 */
public class FactoriaMontana implements FactoriaCarreraYBicicleta {

    @Override
    public Carrera crearCarrera() {
        return new CarreraMontana();
    }

    @Override
    public Bicicleta crearBicicleta() {
        return new BicicletaMontana();
    }
    
}

File: practica1/Ejercicio2/FactoriaMontana.py
from FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta
from carreraMontana import CarreraMontana
from BicicletaMontana import BicicletaMontana

class FactoriaMontana(FactoriaCarreraYBicicleta):
    def crearCarrera(bicicletas):
        return CarreraMontana(bicicletas, porcent_retirada=0)

    def crearBicicleta(id):
        return BicicletaMontana(id)